internal/bminventory/inventory.go

```diff
@@ -279,6 +284,39 @@ func NewBareMetalInventory(
 	}
 }
 
+func (b *bareMetalInventory) updateInfraEnvByID(ctx context.Context, db *gorm.DB, infraEnvID *strfmt.UUID, updates interface{}, streamEvents bool) *gorm.DB {
+	infraenv := models.InfraEnv{
+		ID: infraEnvID,
+	}
+	return b.updateInfraEnv(ctx, db, &infraenv, updates, streamEvents)
+}
+
+func (b *bareMetalInventory) updateInfraEnv(ctx context.Context, db *gorm.DB, infraEnv *models.InfraEnv, updates interface{}, streamEvents bool) *gorm.DB {
```

Comment:
I think this wrapper might not be needed.

It might be better to create a function for sending the event to the stream rather than one for updating the database and sending the event.